### PR TITLE
DO-78 :sparkles: add grafana

### PR DESCRIPTION
**Pull request by dkravetz**
### References

[DO-78](https://xainag.atlassian.net/browse/DO-78)

### Summary

This PR Adds grafana and an initial setup to consume Keycloak for authentication purposes.

### Are there any open tasks/blockers for the ticket?

No

A follow up PR may be necessary to finish the KeyCloak <-> Grafana integration, depending on complexity.

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [ ] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [ ] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [ ] Linked the ticket in the merge request title or the references section.
- [ ] Added an informative merge request summary.

**Code checklist**

- [ ] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [ ] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -70,6 +70,46 @@ services:
...
+      - xain-fl-dev
+    environment:
+      GF_AUTH_GENERIC_OAUTH_ENABLED: "true"
+      GF_AUTH_GENERIC_OAUTH_NAME: "OAuth"
+      GF_AUTH_GENERIC_OAUTH_CLIENT_ID: "grafana"
+      GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET: "c3fbd14d-dfe9-4dc3-9894-8a97bc6c6575"
+      GF_AUTH_GENERIC_OAUTH_ALLOW_SIGN_UP: "true"
+      GF_AUTH_GENERIC_OAUTH_SCOPES: "openid profile email"
+      #GF_AUTH_GENERIC_OAUTH_EMAIL_ATTRIBUTE_NAME: "email:primary"
+      GF_AUTH_GENERIC_OAUTH_AUTH_URL: "http://localhost:8080/auth/realms/master/protocol/openid-connect/auth"
```
Comment (rsaffi) [inline docker-compose-dev.yml]: This should be `keycloak` instead of `localhost`, right?

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -70,6 +70,46 @@ services:
     ports:
       - "8086:8086"
 
+  keycloak:
```
Comment (rsaffi) [inline docker-compose-dev.yml]: Why `keycloak` here? This PR should be about `grafana` only. #283 should be about `keycloak`.

Comment (rsaffi) [reply docker-compose-dev.yml]: I mean, once #283 is merged then this one ideally would be rebased to `development`, or are you just doing two in one shot so you don't have to rebase?

File: docker-compose-dev.yml (hunk 1 above)
Comment (dkravetz) [reply docker-compose-dev.yml]: Nope, this is correct. This is the domain that the browser gets redirected to

File: docker-compose-dev.yml (hunk 2 above)
Comment (dkravetz) [reply docker-compose-dev.yml]: Because this is a branch based off of https://github.com/xainag/xain-fl/pull/283 :D

**Pull request merged**